app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -132,34 +135,39 @@ private void setupFavorListener(View rootView, String favorId) {
                   }
                 }
               } catch (Exception e) {
-                Log.d(TAG, e.getMessage());
+                Log.d(TAG, Objects.requireNonNull(e.getMessage()));
                 CommonTools.showSnackbar(rootView, getString(R.string.error_database_sync));
               }
             });
   }
 
   /** When fragment is launched with favor. */
-  private void displayFavorInfo(View v) {
+  private void displayFavorInfo(View v) throws ExecutionException, InterruptedException {
     favorStatus = FavorStatus.toEnum(currentFavor.getStatusId());
     mTitleView.setText(currentFavor.getTitle());
     mDescriptionView.setText(currentFavor.getDescription());
     String url = currentFavor.getPictureUrl();
     if (url != null) {
-      v.findViewById(R.id.loading_panel).setVisibility(View.VISIBLE);
-      getViewModel()
-          .downloadPicture(currentFavor)
-          .thenAccept(
-              picture -> {
-                mImageView.setImageBitmap(picture);
-                v.findViewById(R.id.loading_panel).setVisibility(View.GONE);
-              });
+      Bitmap pic = getViewModel().loadPictureFromLocal(getContext(), currentFavor).get();
```

Comment:
I think this logic should be placed in the FavorViewModel directly and wrapped inside a single method that returns a picture.

Reply:
Yeah let's look at that in the next sprint. I also need to include this in FavorDetailView.